convert-for-checkers.sh

```diff
@@ -33,33 +34,62 @@ fi | while read -r fAfa; do
     echo "Transforming to .mata"
     echo "@AFA-bits" > $f.mata
     # TODO: is this pipe correct? especially treeRepr? what is treeReprUninit?
-    $LTLE_TO_AFA boomSeparate < $f.afa | $LTLE_TO_AFA removeFinals | $LTLE_TO_AFA treeRepr | $LTLE_TO_AFA initToDnf | $LTLE_TO_AFA prettyToMachete >> $f.mata
+    $AUTOMATA_SAFA_ONE share < $f.afa | $AUTOMATA_SAFA_ONE removeUselessShares | $LTLE_TO_AFA boomSeparate | $LTLE_TO_AFA removeFinals | $LTLE_TO_AFA treeRepr | $LTLE_TO_AFA initToDnf | $LTLE_TO_AFA prettyToMachete >> $f.mata
+  }
+  ${Smv:-false} && {
+    echo "Transforming to .smv"
+    $AUTOMATA_SAFA_ONE share < $f.afa | $AUTOMATA_SAFA_ONE removeUselessShares | $LTLE_TO_AFA prettyToSmv > $f.smv
   }
   ${Aiger:-false} && {
     echo "Transforming to .aig"
-    $LTLE_TO_AFA removeFinalsNonsep < $f.afa | $LTLE_TO_AFA prettyToSmv | $SMVTOAIG > $f.aig
+    $AUTOMATA_SAFA_ONE share < $f.afa | $AUTOMATA_SAFA_ONE removeUselessShares > /tmp/afa.afa
+    if $AUTOMATA_SAFA_ONE hasComplexFinals < /tmp/afa.afa; then
+      $AUTOMATA_SAFA_ONE removeFinals < /tmp/afa.afa | $LTLE_TO_AFA prettyToSmv | $SMVTOAIG > $f.aig
+    else
+      $LTLE_TO_AFA prettyToSmv < /tmp/afa.afa | $SMVTOAIG > $f.aig
+    fi
+  }
+  ${AigerSimp:-false} && {
+    echo "Transforming to .aig"
+
+    mkdir -p /tmp/singleafa{1,2,3}
+
+    $AUTOMATA_SAFA_ONE removeFinals < $f.afa > /tmp/singleafa1/0
+    $LTLE_TO_AFA -i prettyStranger:/tmp/singleafa1 -o afaBasicSimp:/tmp/singleafa2
+    $LTLE_TO_AFA -i afa:/tmp/singleafa2 -o smv:/tmp/singleafa3
+    cat /tmp/singleafa3/0 | $SMVTOAIG > $f.aig
   }
   ${Mona:-false} && {
     echo "Transforming to .mona"
     $LTLE_TO_AFA prettyToMona < $f.afa > $f.mona
   }
   ${Afasat:-false} && {
     echo "Transforming to .afasat"
-    $LTLE_TO_AFA removeFinalsNonsep < $f.afa | $LTLE_TO_AFA prettyToAfasat > $f.afasat
+    $AUTOMATA_SAFA_ONE share < $f.afa | $AUTOMATA_SAFA_ONE removeUselessShares | $AUTOMATA_SAFA_ONE removeFinals | $LTLE_TO_AFA prettyToAfasat > $f.afasat
   }
   ${Ada:-false} && {
     echo "Transforming to .ada"
     $LTLE_TO_AFA prettyToAda < $f.afa > $f.ada 2> /dev/null || \
-    $LTLE_TO_AFA removeFinalsNonsep < $f.afa | $LTLE_TO_AFA prettyToAda > $f.ada
+    $AUTOMATA_SAFA_ONE removeFinalsHind < $f.afa | $LTLE_TO_AFA prettyToAda > $f.ada
```

Comment:
Why did you change this to `removeFinalsHind`? Shouldn't it be `$AUTOMATA_SAFA_ONE removeFinals`

Reply:
yes, thx

Reply:
hind is only good for separated automata (and keeps them separated)